SleepEngine/Engine/Jobs/Event.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+BEGIN_NAMESPACE_SLEEP
+
+class Event
+{
+    using LockGuard = std::lock_guard<std::mutex>;
+public:
+    Event() = default;
+    Event(Event&&) = default;
+
+    void waitAndReset()
+    {
+        std::unique_lock<std::mutex> lk(m_mutex);
+        m_condition.wait(lk, [this] { return m_isSet; });
+        m_isSet = false;
+    }
+
+    void signal()
+    {
+        threadSafeSet(true);
+        m_condition.notify_all();
+    }
+	
+    void reset()
+    {
+        threadSafeSet(false);
+    }
+
+    bool IsSet() const { return m_isSet; }

```

Comment:
Why not GETTER() macro here?

Reply:
I ll refactor them later, to be like PROPERTY or smth, but for now I really don't like them